config/config.exs

```diff
@@ -116,4 +116,6 @@ config :logflare, Logflare.Cluster.Utils, min_cluster_size: 1
 
 config :grpc, start_server: true
 
+config :logflare, Logflare.AlertsScheduler, init_task: {Logflare.Alerting, :init_alert_jobs, []}
```

Comment:
Citrine automatically warms with this `init_task` function when started?

Reply:
What happens when multiple nodes start up and try to add the same job?

Reply:
All these jobs are distributed with mnesia is there a key to unique them or something?

Reply:
yes the `init_task` is triggered for each node startup. using `put_job/1` would update the job completely, including changing the task value.

https://hexdocs.pm/citrine/Citrine.Scheduler.html#c:put_job/1

The `:id` key is used for uniqueness.

Did some digging, and what happens when `put_job/1` is triggered is the distributed process' state will be updated. It would be good if we can avoid the update, as it is linear time complexity to do the update:

https://github.com/brndnmtthws/citrine/blob/master/lib/citrine/job_executor.ex#L158

A few possible strategies is:
1. storing last_init times on db, and having a staleness threshold. The `put_job/1` would only then be executed if it is stale.
2. upstreaming some changes to the library to allow separate of create/update as well as adding in last initialized timestamps to the stored Citrine.Job.

option 2 is more of a long run optimization, and option 1 is likely the simpler way to go. 

Another optimization is use Repo.stream() and startup the alerts in batches. Since this stream would be long running and has linear time complexity, we can add in some checks to ensure that only one node is performing the full table stream of the `init_task`. So that would involve:

- having a genserver that would store the state of whether this initializaiton is taking place
- on startup, the init_task checks across the cluster if an initialization is running. If already running, immediately exit the task. If not, update state and begin the Repo.stream transaction.

Reply:
Task for stalekenss check and streaming https://www.notion.so/supabase/Logflare-Alerts-optimize-init_task-for-loading-Citrine-Jobs-using-staleness-checks-a214a7aa7f1c45929bac25466e1721c9?pvs=4

Reply:
Yeah `last_init` seams reasonable for now.